Mathlib/RingTheory/Polynomial/Chebyshev.lean

```diff
@@ -229,10 +234,185 @@ theorem one_sub_X_sq_mul_U_eq_pol_in_T (n : ℤ) :
   linear_combination T_eq_X_mul_T_sub_pol_U R n
 #align polynomial.chebyshev.one_sub_X_sq_mul_U_eq_pol_in_T Polynomial.Chebyshev.one_sub_X_sq_mul_U_eq_pol_in_T
 
-variable {R S}
+/-- `C n` is the `n`th rescaled Chebyshev polynomial of the first kind (also known as a Vieta–Lucas
+polynomial), given by $C_n(2x) = 2T_n(x)$. See `Polynomial.Chebyshev.C_comp_two_mul_X`. -/
+noncomputable def C : ℤ → R[X]
+  | 0 => 2
+  | 1 => X
+  | (n : ℕ) + 2 => X * C (n + 1) - C n
+  | -((n : ℕ) + 1) => X * C (-n) - C (-n + 1)
+  termination_by n => Int.natAbs n + Int.natAbs (n - 1)
+
+@[simp]
+theorem C_add_two : ∀ n, C R (n + 2) = X * C R (n + 1) - C R n
+  | (k : ℕ) => C.eq_3 R k
+  | -(k + 1 : ℕ) => by linear_combination (norm := (simp [Int.negSucc_eq]; ring_nf)) C.eq_4 R k
+
+theorem C_add_one (n : ℤ) : C R (n + 1) = X * C R n - C R (n - 1) := by
+  linear_combination (norm := ring_nf) C_add_two R (n - 1)
+
+theorem C_sub_two (n : ℤ) : C R (n - 2) = X * C R (n - 1) - C R n := by
+  linear_combination (norm := ring_nf) C_add_two R (n - 2)
+
+theorem C_sub_one (n : ℤ) : C R (n - 1) = X * C R n - C R (n + 1) := by
+  linear_combination (norm := ring_nf) C_add_two R (n - 1)
+
+theorem C_eq (n : ℤ) : C R n = X * C R (n - 1) - C R (n - 2) := by
+  linear_combination (norm := ring_nf) C_add_two R (n - 2)
+
+@[simp]
+theorem C_zero : C R 0 = 2 := rfl
+
+@[simp]
+theorem C_one : C R 1 = X := rfl
+
+theorem C_neg_one : C R (-1) = X := (by ring : X * 2 - X = X)
```

Comment:
Not a big fan of that proof - does something like this work?
```suggestion
theorem C_neg_one : C R (-1) = X := by
  unfold C
  ring
```

Reply:
It doesn't. Why don't you like the proof?

Reply:
How about `show X * 2 - X = x by ring`?